Kickstarter-iOS/Features/PledgePaymentMethods/Controller/NoShippingPledgeRewardsSummaryViewController.swift

```diff
@@ -0,0 +1,314 @@
+      .observeForUI()
+      .observeValues { [weak self] data in
+        guard let self else { return }
+
+        self.applySnapshotToDataSource(from: data)
+
+        self.setEntireViewToIsHidden(false)
+        self.tableViewContainerHeightConstraint?.constant = self.tableView.intrinsicContentSize.height
+      }
+
+    self.viewModel.outputs.configurePledgeTotalViewWithData
+      .observeForUI()
+      .observeValues { [weak self] pledgeSummaryData in
+        guard let self else { return }
+
+        self.pledgeTotalViewController.configure(with: pledgeSummaryData)
+      }
+  }
+
+  // MARK: - Configuration
+
+  func configureWith(
+    rewardsData: PostCampaignRewardsSummaryViewData,
+    bonusAmount: Double?,
+    pledgeData: PledgeSummaryViewData
+  ) {
+    self.viewModel.inputs
+      .configureWith(rewardsData: rewardsData, bonusAmount: bonusAmount, pledgeData: pledgeData)
+
+    self.view.setNeedsLayout()
+  }
+
+  // MARK: Styles
+
+  private func tableViewStyle(_ tableView: UITableView) {
+    tableView.separatorInset = .zero
+    tableView.contentInsetAdjustmentBehavior = .never
+    tableView.isScrollEnabled = false
+    tableView.delegate = self
+    tableView.rowHeight = UITableView.automaticDimension
+    tableView.separatorStyle = .none
+    tableView.backgroundColor = .ksr_white
+    tableView.translatesAutoresizingMaskIntoConstraints = false
+  }
+
+  private func rootStackViewStyle(_ stackView: UIStackView) {
+    stackView.axis = NSLayoutConstraint.Axis.vertical
+    stackView.spacing = Styles.grid(1)
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.translatesAutoresizingMaskIntoConstraints = false
+  }
+
+  private func separatorViewStyle(_ view: UIView) {
+    view.backgroundColor = .ksr_support_200
+    view.translatesAutoresizingMaskIntoConstraints = false
+  }
+
+  private func sectionHeaderViewStyle(_ view: UIView) {
+    view.translatesAutoresizingMaskIntoConstraints = false
+    view.frame = CGRectMake(0, 0, self.tableView.frame.size.width, self.tableView.frame.size.height)
+  }
+
+  private func sectionHeaderLabelStyle(_ label: UILabel) {
+    label.font = UIFont.ksr_subhead().bolded
+    label.textColor = UIColor.ksr_black
+    label.numberOfLines = 0
+    label.frame = CGRectMake(
+      CheckoutConstants.PledgeView.Inset.leftRight,
+      0,
+      self.tableView.frame.size.width,
+      PledgeRewardsSummaryStyles.Layout.sectionHeaderLabelHeight
+    )
+  }
+
+  private func tableViewContainerStyle(_ view: UIView) {
+    view.translatesAutoresizingMaskIntoConstraints = false
+    view.clipsToBounds = true
+  }
+
+  // MARK: - Helpers
+
+  private func setEntireViewToIsHidden(_ isHidden: Bool) {
+    self.view.isHidden = isHidden
+    self.pledgeTotalViewController.view.isHidden = isHidden
+    self.separatorView.isHidden = isHidden
+  }
+
+  // MARK: Apply DataSource Snapshot
+
+  private func applySnapshotToDataSource(from data: [PostCampaignRewardsSummaryItem]) {
```

Comment:
This method defines the sections and data used in the table via `snapshot.appendSections` and `snapshot.appendItems`, respectively. Then, it updates the table all at once using `apply.`